### PR TITLE
feat: support sitemap

### PR DESCRIPTION
**Pull request by Innei**
### <samp>🤖 Generated by Copilot at 2784a75</samp>

This pull request adds support for generating and serving dynamic sitemap.xml files for each site in the Crossbell-Box/xLog app. It also improves the `NextServerResponse` class to handle custom headers and text responses, and cleans up some unused code in the `middleware.ts` file.

### <samp>🤖 Generated by Copilot at 2784a75</samp>

> _We are the masters of the sitemap_
> _We fetch the data with the `queryClient`_
> _We set the headers and the text response_
> _We bypass the middleware with our might_

### WHY

Support sitemap in app router

### <samp>🤖 Generated by Copilot at 2784a75</samp>

*  Add a new route handler for the sitemap.xml file for each site ([link](https://github.com/Crossbell-Box/xLog/pull/559/files?diff=unified&w=0#diff-87e647d89a99a35f053f98e965b14656de85c5aca19532be779dc67410624862R1-R58))
*  Refactor the `NextServerResponse` class in `server-helper.ts` to support custom headers ([link](https://github.com/Crossbell-Box/xLog/pull/559/files?diff=unified&w=0#diff-59102566004aa11373940ed74936c88f94b12fe12bce3910598cdc2f553ae738R18), [link](https://github.com/Crossbell-Box/xLog/pull/559/files?diff=unified&w=0#diff-59102566004aa11373940ed74936c88f94b12fe12bce3910598cdc2f553ae738L28-R31), [link](https://github.com/Crossbell-Box/xLog/pull/559/files?diff=unified&w=0#diff-59102566004aa11373940ed74936c88f94b12fe12bce3910598cdc2f553ae738L45-R61), [link](https://github.com/Crossbell-Box/xLog/pull/559/files?diff=unified&w=0#diff-59102566004aa11373940ed74936c88f94b12fe12bce3910598cdc2f553ae738R85-R91))
*  Remove the condition to bypass the custom middleware for the sitemap.xml file in `middleware.ts` ([link](https://github.com/Crossbell-Box/xLog/pull/559/files?diff=unified&w=0#diff-0626369278c89781505d2c04865ecce1302e979271bcfb471e029c19addc13a0L59-R59))

### CLAIM REWARDS

For first-time contributors, please leave your xLog address and Discord ID below to claim your rewards.

### CODE REVIEW

File: src/app/site/[site]/sitemap.xml/route.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+  const queryClient = new QueryClient()
+  // http://localhost:2222/site/innei-4525/sitemap.xml
+  // http://innei-4525.localhost:2222/sitemap.xml
+  // find innei-5425 site handle by regexp
+  const domainOrSubdomain =
+    req.nextUrl.pathname.match(/\/site\/(.*)\/sitemap.xml/)?.[1] ||
+    // http://innei-4525.localhost:2222/sitemap.xml
+    req.headers.get("host")?.split(".")[0] ||
+    ""
```
Comment (DIYgod) [inline src/app/site/[site]/sitemap.xml/route.ts:23]: It seems that the custom domain is not functioning properly. I suggest adding a header `x-handle` to the middleware https://github.com/Crossbell-Box/xLog/blob/dev/src/middleware.ts#L86 and retrieving the handle from the header.

Comment (Innei) [reply src/app/site/[site]/sitemap.xml/route.ts:23]: I tested `http://innei-4525.localhost:2222/sitemap.xml` before and there was no problem. Does it refer to other meanings? I don't understand.

Comment (DIYgod) [reply src/app/site/[site]/sitemap.xml/route.ts:23]: I am referring to custom domains like `https://blog.innei.ren/`.

File: src/middleware.ts [hunk 2]
```diff
@@ -84,6 +83,7 @@ export default async function middleware(req: NextRequest) {
   const requestHeaders = new Headers(req.headers)
   requestHeaders.set("x-pathname", req.nextUrl.pathname)
   requestHeaders.set("x-search", req.nextUrl.search)
+  requestHeaders.set("x-handle", tenant.subdomain || "")
```
Comment (Innei) [inline src/middleware.ts:86]: Add this? Is it can works? @DIYgod

**Comment by DIYgod** [comment]
Thanks so much!

**Pull request merged**